components/connection_protocols/tcp.js

```diff
@@ -113,7 +114,7 @@ class TCPConnection extends BaseConnection {
 
 		this.stream.on('timeout', () => {
 			this._debug('TCP connection timed out');
-			this.user._connectTimeout = Math.min(this.user._connectTimeout * 2, 10000); // 10 seconds max
+			this.user._connectTimeout = Math.min(this.user._connectTimeout * 2, this.maxTimeout); // X seconds max
```

Comment:
are we absolutely sure that changing timeout here is a good idea?

Reply:
I mean it will just come in to use when actually timing out. The initial connection timeout will be still 1000 as before https://github.com/DoctorMcKay/node-steam-user/blob/master/index.js#L48. It's just that the highest possible connection timeout after reconnecting will be whatever the user decides on.